vespalib/src/vespa/vespalib/util/cpu_usage.h

```diff
@@ -7,6 +7,19 @@ namespace vespalib {
 
 namespace cpu_usage {
 
+// do not use this directly (use ThreadSampler)
+class DummyThreadSampler {
+private:
+    steady_time _start;
+    double _load;
+public:
+    DummyThreadSampler();
+    void expected_load(double load);
+    duration sample() const;
+};
+
+#ifdef __linux__
```

Comment:
Is it possible to do this in the cpp file only to avoid ifdefs in the header files.

Reply:
yes, but it will have a cost

Reply:
what about an is_linux.h that defines a constexpr bool that we can use?